docs/design/configuration.md

```diff
@@ -0,0 +1,141 @@
+# Kubernetes Component Configuration
+
+## Abstract
+
+A generic ``Configuration`` object to store command-line flags that is used for
```

Comment:
The example of running OpenShift on Kubernetes uses a Secret for this purpose, and I would like to only use the Secret to hold secret information and not all general config.  https://github.com/GoogleCloudPlatform/kubernetes/tree/master/examples/openshift-origin